basyx.aasrepository/basyx.aasrepository-http/src/test/java/org/eclipse/digitaltwin/basyx/aasrepository/http/AasRepositoryHTTPSuite.java

```diff
@@ -283,6 +298,58 @@ public void paginationResult() throws FileNotFoundException, IOException, ParseE
 		BaSyxHttpTestUtils.assertSameJSONContent(getPaginatedAas1JSONString(), getJSONWithoutCursorInfo(response));
 	}
 
+	@Test
+	public void uploadThumbnailToShell() throws IOException {
+		createDummyAasOnServer(getAas1JSONString());
+		CloseableHttpResponse getThumbnailResponse = uploadThumbnail(dummyAasId);
+
+		assertEquals(HttpStatus.OK.value(), getThumbnailResponse.getCode());
+	}
+
+	@Test
+	public void getFile() throws FileNotFoundException, IOException, ParseException {
+		createDummyAasOnServer(getAas1JSONString());
+
+		byte[] expectedFile = readBytesFromClasspath(THUMBNAIL_FILE_PATH);
+
+		uploadThumbnail(dummyAasId);
+
+		CloseableHttpResponse response = BaSyxHttpTestUtils.executeGetOnURL(getThumbnailAccessURL(dummyAasId));
+		assertEquals(HttpStatus.OK.value(), response.getCode());
+
+		byte[] actualFile = EntityUtils.toByteArray(response.getEntity());
+
+		response.close();
+
+		assertArrayEquals(expectedFile, actualFile);
+	}
+
+	@Test
+	public void getFileFromNotExistElement() throws FileNotFoundException, UnsupportedEncodingException, ClientProtocolException, IOException {
+		CloseableHttpResponse response = BaSyxHttpTestUtils.executeGetOnURL(getThumbnailAccessURL(dummyAasId));
+
+		assertEquals(HttpStatus.NOT_FOUND.value(), response.getCode());
+	}
+
+	@Test
+	public void deleteFile() throws FileNotFoundException, IOException {
+		createDummyAasOnServer(getAas1JSONString());
+		uploadThumbnail(dummyAasId);
+
+		CloseableHttpResponse response = BaSyxHttpTestUtils.executeDeleteOnURL(getThumbnailAccessURL(dummyAasId));
+		assertEquals(HttpStatus.OK.value(), response.getCode());
+
+		response = BaSyxHttpTestUtils.executeGetOnURL(getThumbnailAccessURL(dummyAasId));
```

Comment:
Separate in another test

Reply:
This cannot be seperated since I at first create a thumbnail because the shell had no thumbnail before, then I delete it, after that I use to get thumbnail to validate that the thumbnail is deleted.

Reply:
clarified